specification/appconfiguration/AppConfiguration/routes.tsp

```diff
@@ -0,0 +1,973 @@
+https://aka.ms/azconfig/docs/keyvaluefiltering
+""")
+    @query("key")
+    key?: string;
+
+    @doc("""
+A filter used to match labels. Syntax reference:
+https://aka.ms/azconfig/docs/keyvaluefiltering
+""")
+    @query("label")
+    label?: string;
+
+    ...syncTokenHeader;
+
+    @doc("""
+Instructs the server to return elements that appear after the element referred
+to by the specified token.
+""")
+    @query("After")
+    after?: string;
+
+    @doc("""
+Requests the server to respond with the state of the resource at the specified
+time.
+""")
+    @header("Accept-Datetime")
+    acceptDatetime?: string;
+
+    @doc("Used to select what fields are present in the returned resource(s).")
+    @query({
+      name: "$Select",
+      format: "csv",
+    })
+    select?: KeyValueFields[];
+
+    @doc("A filter used get key-values for a snapshot. Not valid when used with 'key' and 'label' filters.")
+    @query("snapshot")
+    snapshot?: string;
+
+    @doc("""
+Used to perform an operation only if the targeted resource's etag matches the
+value provided.
+""")
+    @header("If-Match")
+    ifMatch?: string;
+
+    @doc("""
+Used to perform an operation only if the targeted resource's etag does not
+match the value provided.
+""")
+    @header("If-None-Match")
+    ifNoneMatch?: string;
+
+    #suppress "@azure-tools/typespec-azure-core/prefer-csv-collection-format" "Pre-existing API contract"
+    @doc("""
+A filter used to query by tags. Syntax reference:
+https://aka.ms/azconfig/docs/keyvaluefiltering
+""")
+    @query({
+      name: "tags",
+      format: "multi",
+    })
+    tags?: string[];
+  },
+  OkResponse & AppConfigResponseHeaders,
+  {},
+  Error
+>;
+
+@summary("Gets a single key-value.")
+@doc("Gets a single key-value.")
+op getKeyValue is standardOps.ResourceRead<
+  KeyValue,
+  QueryParametersTrait<{
+    @doc("The label of the key-value to retrieve.")
+    @query("label")
+    label?: string;
+
+    @doc("Used to select what fields are present in the returned resource(s).")
+    @query({
+      name: "$Select",
+      format: "csv",
+    })
+    select?: KeyValueFields[];
+  }> &
+    RequestHeadersTrait<{
+      ...syncTokenHeader;
+
+      @doc("""
+Requests the server to respond with the state of the resource at the specified
+time.
+""")
+      @header("Accept-Datetime")
+      acceptDatetime?: string;
+
+      @doc("""
+Used to perform an operation only if the targeted resource's etag matches the
+value provided.
+""")
+      @header("If-Match")
+      ifMatch?: string;
+
+      @doc("""
+Used to perform an operation only if the targeted resource's etag does not
+match the value provided.
+""")
+      @header("If-None-Match")
+      ifNoneMatch?: string;
+    }> &
+    ResponseHeadersTrait<AppConfigResponseHeaders &
+      contentTypeHeader<"application/vnd.microsoft.appconfig.kv+json">>
+>;
+
+// FIXME: check is service returns 201 and/or 200?
+// Check if this operation is an LRO?
```

Comment:
@avanigupta could you please answer these questions for this operation?

Reply:
`PUT /snapshot` is the only LRO (and the only operation to return 201).